### PR TITLE
FUSETOOLS-2092 - Fix NPE In Validation

### PR DESCRIPTION
**Pull request by apupier**
- avoid NPE when the service interface class is not available on the
project classpath
- update test: 3 others validation issues were hidden before

--> @bfitzpat Can you ensure that the new validation error are correct from a SY point of view?

next step will be to provide a more precise test to check for exact expected exception errors and not only the global number. (but might be done in another PR)

### CODE REVIEW

**Comment by bfitzpat** [comment]
Looking at this now

**Comment by bfitzpat** [comment]
I have updated FUSETOOLS-2092 with details on ALL of the errors and warnings that are reported in the validator project. The only one I'm not sure of yet is "No interface is defined for the service/reference." and I'm trying to get more details from Tom & Tomo to understand it.

**Comment by bfitzpat** [comment]
I am thinking something is off about the validation logic here - https://github.com/jboss-switchyard/tools/blob/master/eclipse/plugins/org.switchyard.tools.ui/src/org/switchyard/tools/ui/validation/ServiceInterfaceConstraint.java#L63

**Comment by bfitzpat** [comment]
But I think that all of the other warnings & errors are correct. Just wondering about these last two (which are really both in the same validation logic)

**Comment by bfitzpat** [comment]
I put a solution to the validation logic issue into the JIRA FUSETOOLS-2092 - but I was unable to get a branch to work on top of this one. Perhaps you can help me with that in the morning.

**Comment by apupier** [comment]
ok to help on rebasing the branch.

File: eclipse/plugins/org.switchyard.tools.ui/src/org/switchyard/tools/ui/validation/WiringValidationContext.java [hunk 1]
```diff
@@ -262,8 +262,10 @@ private void collectServiceInterfaces(SwitchYardType switchyard) {
                             _serviceInterfaces.put((Contract) eobject, si);
                         }
                     } catch (Exception e) {
+                        e.printStackTrace();
```
Comment (lhein) [inline eclipse/plugins/org.switchyard.tools.ui/src/org/switchyard/tools/ui/validation/WiringValidationContext.java]: print stacktrace? Why not log into error log?

**Comment by lhein** [comment]
+1

**Comment by apupier** [comment]
testPR

**Comment by bfitzpat** [comment]
Clean PR build

**Pull request merged**